tests/integration/components/polaris-resource-list/filter-control/filter-creator-test.js

```diff
@@ -0,0 +1,468 @@
+      });
+
+      await render(hbs`
+        {{polaris-resource-list/filter-control/filter-creator
+          filters=filters
+          resourceName=resourceName
+          disabled=disabled
+        }}
+      `);
+
+      assert.dom('.Polaris-Popover').doesNotExist();
+    });
+
+    test('renders a active popover with a Select on click of the activator button', async function(assert) {
+      this.setProperties({
+        filters,
+        resourceName,
+        disabled: false,
+      });
+
+      await render(hbs`
+        {{polaris-resource-list/filter-control/filter-creator
+          filters=filters
+          resourceName=resourceName
+          disabled=disabled
+        }}
+      `);
+
+      await activatePopover();
+
+      assert.dom('.Polaris-Popover').exists();
+    });
+
+    test('renders a non-active popover after add filter button was clicked and onAddFilter was triggered', async function(assert) {
+      this.setProperties({
+        filters,
+        resourceName,
+        disabled: false,
+      });
+
+      await render(hbs`
+        {{polaris-resource-list/filter-control/filter-creator
+          filters=filters
+          resourceName=resourceName
+          disabled=disabled
+          onAddFilter=(action (mut wasOnAddFilterCalled) true)
+        }}
+      `);
+
+      await activatePopover();
+      await selectFilterKey(filters[0].key);
+      await selectFilterValue('Bundle');
+      await clickAddFilter();
+
+      assert.ok(this.get('wasOnAddFilterCalled'));
+      assert.dom('.Polaris-Popover').doesNotExist();
+    });
+
+    test('does not renders FilterValueSelector after add filter button was clicked', async function(assert) {
+      this.setProperties({
+        filters,
+        resourceName,
+        disabled: false,
+      });
+
+      await render(hbs`
+        {{polaris-resource-list/filter-control/filter-creator
+          filters=filters
+          resourceName=resourceName
+          disabled=disabled
+          onAddFilter=(action (mut wasOnAddFilterCalled) true)
+        }}
+      `);
+
+      await activatePopover();
+      await selectFilterKey(filters[0].key);
+      await selectFilterValue('Bundle');
+
+      assert.ok(filterValueSelector);
+      await clickAddFilter();
+      assert.notOk(filterValueSelector);
+    });
+
+    skip('renders Select with no value after add filter button was clicked', async function(assert) {
+      this.setProperties({
+        filters,
+        resourceName,
+        disabled: false,
+      });
+
+      await render(hbs`
+        {{polaris-resource-list/filter-control/filter-creator
+          filters=filters
+          resourceName=resourceName
+          disabled=disabled
+          onAddFilter=(action (mut wasOnAddFilterCalled) true)
+        }}
+      `);
+
+      await activatePopover();
+      await selectFilterKey(filters[0].key);
+      await selectFilterValue('Bundle');
+
+      assert.dom('[data-test-select="filter"]').hasAnyValue();
```

Comment:
I think this is borking here because `hasAnyValue` eventually makes its way into the `hasValue` assertion under the hood, where it grabs the target using [`element as HTMLInputElement`](https://github.com/simplabs/qunit-dom/blob/master/lib/assertions.ts#L696). In this test we're grabbing `'[data-test-select="filter"]'` which is an `HTMLSelectElement` type.

So something like

```javascript
let filterSelectValue = find('[data-test-select="filter"]').value;

assert.ok(filterSelectValue);
```

would probably work here instead?

Reply:
(and then same for the next assertion)

Reply:
aw bummer 😞 ok then.